docs/source/docker/integration/docker_with_datasource.rst

```diff
@@ -0,0 +1,139 @@
+
+.. _ref-docker-with-datasource:
+
+Using the docker with an S3 bucket as remote datasource.
+========================================================
+
+Introduction
+------------
+The Lightly Docker can be used with the Lightly Platform to do
+the following workloads in one single run:
+
+- stream your files directly from your AWS S3 bucket to your local machine without
+  needing to sync or download them
+- embed all images or video frames
+- sample a subset, e.g. using CORESET
+- compute the metadata of the images
+- create a dataset in the Lightly Platform from the sampled subset
+
+Support for the following is planned but not implemented yet:
+
+- Training an embedding model is currently too slow and thus deactivated.
+- Streaming from a google cloud storage or azure storage is planned to be implemented.
+
+If you need any of these, write us so that we prioritize implementing it.
+
+Advantages
+----------
+
+- You can run the whole Lightly workflow with one single command.
+- You can process videos directly without needing to extract frames.
+- Your data is streamed from your S3 bucket on the go instead of being first downloaded
+  to your local disk and then read from there. Thus you save both a lot of time and
+  disk space and can process much larger datasets.
+- You can :ref:`ref-docker-with-datasource-datapool`
+  to continuously update and improve your subsampled dataset
+  everytime new data comes into your S3 bucket.
+- Your images and videos are never saved anywhere but in your S3 bucket,
+  maintaining your privacy and security.
+
+
+Requirements
+------------
+
+This recipe requires that you already have a dataset in the Lightly Platform
+configured to use the data in your AWS S3 bucket.
+
+If you don`t have such a dataset yet,
+follow the steps on how to
+`create a Lightly dataset connected to your S3 bucket <https://docs.lightly.ai/getting_started/dataset_creation/dataset_creation_aws_bucket.html>`_.
+
+Furthermore, you should have access to a machine running docker.
+Ideally, it also has a CUDA-GPU.
+A fast GPU will speed up the process significantly,
+especially for large datasets.
+
+
+Download the Lightly Docker
+---------------------------------------------
+Next, the Lightly Docker should be installed.
+Please follow the instructions for the :ref:`ref-docker-setup`.
+
+
+Run the Lightly Docker with the datasource
+------------------------------------------
+Head to the :ref:`rst-docker-first-steps` to get a general idea of what the docker
+can do.
+
+For running the docker with a remote datasouce,
+use the parameter `datasource.dataset_id=YOUR_DATASET_ID`.
+You find the dataset id in the Lightly Platform.
+E.g. run the docker with
+
+.. code-block:: console
+
+    docker run --gpus all --rm -it \
```

Comment:
I would try to give a more complete docker run command. Maybe add the stopping condition?